src/parse.rs

```diff
@@ -1764,4 +1764,35 @@ mod test {
         }
         assert_eq!(found, 1);
     }
+
+    #[test]
+    fn common_lifetime_patterns_allowed<'b>() {
+        let temporary_str = String::from("xyz");
+
+        // NOTE: this is a limitation of Rust, it doesn't allow putting lifetime parameters on the closure itself.
+        // Hack it by attaching the lifetime to the test function instead.
+        // TODO: why is the `'b` lifetime required at all? Changing it to `'_` breaks things :(
+        let mut closure = |link: BrokenLink<'b>| {
+            Some(("#".into(), link.reference.into()))
+        };
+
+        fn function<'a>(link: BrokenLink<'a>) -> Option<(CowStr<'a>, CowStr<'a>)> {
+            Some(("#".into(), link.reference))
+        }
+
+        for _ in Parser::new_with_broken_link_callback("static lifetime", Options::empty(), Some(&mut closure)) {
+        }
+        /* This fails to compile. Because the closure can't say `for <'a> fn(BrokenLink<'a>) ->
+         * CowStr<'a>` and has to use the enclosing `'b` lifetime parameter, `temporary_str` lives
+         * shorter than `'b`. I think this is unlikely to occur in real life, and if it does, the
+         * fix is simple: move it out to a function that allows annotating the lifetimes.
+         */
+        //for _ in Parser::new_with_broken_link_callback(&temporary_str, Options::empty(), Some(&mut callback)) {
+        //}
```

Comment:
Let me know if I should make this a `compile_fail` doctest instead; I'm not sure whether this is a little too much to hit people with at once, but it would make sure that this is tested.

Reply:
I think we can do without. We're not really testing a feature of the library by ensuring that that doesn't compile.